Add description for the Built-In Quality practices

aatwi opened the pull request:
1. Create a new MD file (Topics.md)
2. Add description for the TDD practice

Commit:
Add description for the Built-In Quality practices

1. Create a new MD file (Topics.md)
2. Add description for the TDD practice

philou commented on Topics.md:
```diff
@@ -0,0 +1,20 @@
+* Write production code that makes that test pass. 
+* Clean up the mess you just made.
+
+
```
What are the top 3 advantages of TDD regarding built-in quality according to you?

philou commented:
Do you know what IP constraints on the image file?

philou reviewed (commented):
Also, do you know if there are any IP constraints on the image?

Commit:
Add description file for the Built-In Quality practices

1. Rename file from Topics to Practices
2. Add 3 advantages of using TDD
3. Add a reference to the TDD image
